packages/block-editor/src/components/use-on-block-drop/index.js

```diff
@@ -44,75 +45,83 @@ export function parseDropEvent( event ) {
 /**
  * A function that returns an event handler function for block drop events.
  *
- * @param {string}   targetRootClientId        The root client id where the block(s) will be inserted.
- * @param {number}   targetBlockIndex          The index where the block(s) will be inserted.
+ * @param {string} targetRootClientId        The root client id where the block(s) will be inserted.
+ * @param {number} targetBlockIndex          The index where the block(s) will be inserted.
  * @param {Function} getBlockIndex             A function that gets the index of a block.
  * @param {Function} getClientIdsOfDescendants A function that gets the client ids of descendant blocks.
  * @param {Function} moveBlocksToPosition      A function that moves blocks.
- *
+ * @param {Function} insertBlocks              A function that inserts blocks.
+ * @param {Function} clearSelectedBlock        A function that clears block selection.
  * @return {Function} The event handler for a block drop event.
  */
 export function onBlockDrop(
 	targetRootClientId,
 	targetBlockIndex,
 	getBlockIndex,
 	getClientIdsOfDescendants,
-	moveBlocksToPosition
+	moveBlocksToPosition,
+	insertBlocks,
+	clearSelectedBlock
 ) {
 	return ( event ) => {
 		const {
 			srcRootClientId: sourceRootClientId,
 			srcClientIds: sourceClientIds,
 			type: dropType,
+			blocks,
 		} = parseDropEvent( event );
 
-		// If the user isn't dropping a block, return early.
-		if ( dropType !== 'block' ) {
-			return;
+		// If the user is inserting a block
+		if ( dropType === 'inserter' ) {
+			clearSelectedBlock();
+			insertBlocks( blocks, targetBlockIndex, targetRootClientId, false );
```

Comment:
The buttons block seems to have some kind of autofocus, which overrides the `false`.

Because of this perhaps?
https://github.com/WordPress/gutenberg/blob/694fe1114100747d154d78e5df263c4f9ef1873c/packages/block-library/src/buttons/edit.js#L43

Reply:
yes, that block is acting weird, we should address it separately though because I'm not sure about the consequences of such change.